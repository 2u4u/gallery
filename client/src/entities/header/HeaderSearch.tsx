import React, { useRef, useState } from 'react';
import { useOutsideClick } from 'shared/hooks/useOutsideClick';
import { FileMimeType } from 'shared/types/index.type';

type Props = {
    handleSearch: (e: React.ChangeEvent<HTMLInputElement>) => void;
    handleTypeSelect: (type?: FileMimeType) => void;
}

function HeaderSearch({ handleTypeSelect, handleSearch }: Props) {
    const wrapperRef = useRef<HTMLInputElement>(null);
    const [searchType, setSearchType] = useState<FileMimeType>();
    const [selectOpen, setSelectOpen] = useState(false);

    const handleToggleSelect = () => {
        setSelectOpen(selectOpen => !selectOpen);
    }

    const handleCloseSelect = () => {
        setSelectOpen(false);
    }

    useOutsideClick(wrapperRef, handleCloseSelect);

    const handleTypeClick = (type?: FileMimeType) => {
        handleTypeSelect(type);
        setSearchType(type);
        setSelectOpen(false);
    }

    return (
        <div className='header__search search'> 
            <input id='search' type="text" className='search__input' placeholder='Search' onChange={handleSearch}/>
            <label htmlFor="search" className='search__label'>
                <img src="./img/search.svg" alt="search" />
            </label>
            <div ref={wrapperRef} className='search__filter filter'>
                <div onClick={handleToggleSelect} className='filter__block'>
                    <span className='filter__text'>{searchType}</span>
                    {selectOpen 
                        ? (<button type='button' className='filter__button'>
                            <img src="./img/arrow-up.svg" alt="filter" />
                        </button>)
                        : (<button type='button' className='filter__button'>
                            <img src="./img/arrow-down.svg" alt="filter" />
                        </button>)
                    }
                </div>
                {selectOpen ? (
                    <div className='filter__content'>
                        <button type='button' className='filter__item' onClick={() => handleTypeClick()}><i>--all--</i></button>
                        <button type='button' className='filter__item' onClick={() => handleTypeClick('image')}>image</button>
                        <button type='button' className='filter__item' onClick={() => handleTypeClick('audio')}>audio</button>
                        <button type='button' className='filter__item' onClick={() => handleTypeClick('video')}>video</button>
                    </div>
                ) : null}
            </div>
        </div>
    );
}

export default HeaderSearch;
