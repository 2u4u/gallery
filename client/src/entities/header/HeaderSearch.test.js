import renderer from 'react-test-renderer';
import HeaderSearch from './HeaderSearch';

describe('Header search component ', () => {
    it('matches snapshot', () => {
        const handleTypeSelect = () => false;
        const handleSearch = () => false;
        const component = renderer.create(
            <HeaderSearch 
                handleTypeSelect={handleTypeSelect}
                handleSearch={handleSearch}
            />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
