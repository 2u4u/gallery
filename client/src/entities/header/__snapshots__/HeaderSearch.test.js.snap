// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header search component  matches snapshot 1`] = `
<div
  className="header__search search"
>
  <input
    className="search__input"
    id="search"
    onChange={[Function]}
    placeholder="Search"
    type="text"
  />
  <label
    className="search__label"
    htmlFor="search"
  >
    <img
      alt="search"
      src="./img/search.svg"
    />
  </label>
  <div
    className="search__filter filter"
  >
    <div
      className="filter__block"
      onClick={[Function]}
    >
      <span
        className="filter__text"
      />
      <button
        className="filter__button"
        type="button"
      >
        <img
          alt="filter"
          src="./img/arrow-down.svg"
        />
      </button>
    </div>
  </div>
</div>
`;
